autogluon/inference/docker/0.6/py3/cu112/Dockerfile.gpu

```diff
@@ -0,0 +1,58 @@
+ARG PYTHON_VERSION=3.8.10
+
+FROM 763104351884.dkr.ecr.us-west-2.amazonaws.com/mxnet-inference:1.9.0-gpu-py38-cu112-ubuntu20.04-sagemaker
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+# PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK - see https://github.com/autogluon/autogluon/issues/2534
+ENV MXNET_CUDNN_LIB_CHECKING=0 \
+    MXNET_CUDNN_AUTOTUNE_DEFAULT=0 \
+    PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK=1
+
+RUN apt-get update \
+ && apt-get -y upgrade \
+ && apt-get autoremove -y \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+ARG AUTOGLUON_VERSION=0.6.1
+ARG TORCH_VER=1.12.1
+ARG TORCH_VISION_VER=0.13.1
+ARG TORCH_TEXT_VER=0.13.1
+
+# NOTE: reinstalling horovod with pytorch support because it is available in the container
+# Updating certifi - https://pyup.io/p/pypi/certifi/2ff/
+RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.pythonhosted.org pip \
+ && pip install --no-cache-dir -U setuptools wheel \
+ && pip install --no-cache-dir -U certifi \
+ && pip uninstall -y dataclasses \
+ && pip install --no-cache-dir -U numpy numba \
+ && pip install --no-cache-dir -U torch==${TORCH_VER} torchvision==${TORCH_VISION_VER} torchtext==${TORCH_TEXT_VER} --extra-index-url https://download.pytorch.org/whl/cu113 \
+ && pip install --no-cache-dir -U autogluon==${AUTOGLUON_VERSION} \
+ && mim install mmcv-full \
+ && pip install --no-cache-dir -U mmdet
+
+# Downgrade protobuf 3.20.1 -> 3.19.6: https://pyup.io/v/51167/f17
+RUN pip install -U protobuf==3.19.*
```

Comment:
This downgrade seems to be unnecessary to v0.6.1, see https://github.com/autogluon/autogluon/pull/2193

Reply:
It is necessary because the existing version range in autogluon is not supporting safe version of 3.20.x (which is 3.20.6, I believe)

Reply:
Just a quick note per offline discussion, it is the requirement from sagemaker-training 4.3.1 that requires protobuf=3.19.x
```
sagemaker-training 4.3.1 requires protobuf<3.20,>=3.9.2, but you have protobuf 4.21.12 which is incompatible.
```